### DISCUSSION TITLE
Specifying what task must run after another task

### DISCUSSION THREAD

**Discussion by jakubmazanec** [vercel/turborepo#5368] [Help]
Imagine I have packages in a monorepo that have two tasks:

```json
{
  "scripts": {
    "format": "prettier",
    "document": "typedoc"
  }
}
```

Is there a way to configure Turborepo so it runs `format` task also after `document` task? Because formatting must be always done after generating documentation, but not vice versa, so I can't use the usual `dependsOn` field in `turbo.json`. Documentation also doesn't seem to help, so maybe this functionality isn't supported? Thanks for the help.

**Comment by mehulkar** [comment]
The way to configure this is to make `format` dependOn `document`:

```json
{
    "pipeilne": {
        "format": {
            "dependsOn": ["document"]
        },
        "document": {}
    }
}
```

I agree that it feels weird, to configure it like this though.

**Reply by jakubmazanec** [reply]
Thanks for the reply, but then if I want to run `format`, it will always also run `document`, that's not what I want.

**Reply by gsoltis** [reply]
@jakubmazanec if you do need to ensure that `document` has been run before `format`, the dependency is the correct way to do it, you do in fact have a dependency. To avoid rerunning `document` unnecessarily, take advantage of [caching](https://turbo.build/repo/docs/core-concepts/caching)

**Reply by jakubmazanec** [reply]
That's not what I need. I need to ensure that `format` task is run after `document` task. The thing is,`format` is a task that is sometimes used on its own - you simply want to ensure your files are formatted correctly - so it can't have any dependencies (I don't want to run `document` needlessly, and caching will not help with that).
`document` task (i.e. generating documentation) produces files that are sometimes not formatted correctly, that's the reason why I would like that `format` is run after `document` automatically. I could make the `document` task itself contain commands that format the files, I just thought that it would be nice if I could reuse existing task. But it seems that we can describe the realtion between some two tasks only in one direction (via `dependsOn` property), but not the other way around, i.e. what task is depending on this task.

**Comment by maxim-grishaev** [comment]
Bump this. I have basically the same case: generate api, format after. But I can run format without format too.
How to achieve this? 
I was thinking "[with](https://turbo.build/repo/docs/reference/configuration#with)" would help, but 
```
turbo 2.4.4

turbo_json_parse_error

  × Failed to parse turbo.json.
  ├─▶   × Found an unknown key `with`.
```

**Reply by chris-olszewski** [reply]
`with` would not help your use case as `with` doesn't provide any ordering so the formatter could run before the generation finishes.
